client/me/controller.js

```diff
@@ -86,3 +91,41 @@ export function mcpSetup( context, next ) {
 	);
 	next();
 }
+
+// Helper thunk that ensures that the user preferences has been fetched into Redux state before we
+// continue working with it.
+const waitForPrefs = () => async ( dispatch, getState ) => {
```

Comment:
As you can see I've copy pasted a lot of code. Given this is all temporary stuff that only exists as long as the roll out is happening, I wondered whether it is cleaner to copy-paste rather than find an obvious place for all this. And it will be easy to clean up.

But I could be wrong.